app/views/users/new.html.haml

```diff
@@ -1,7 +1,13 @@
+<script type="text/javascript" src="https://js.stripe.com/v2/"></script>
+
+<script type="text/javascript">Stripe.setPublishableKey("#{ENV['STRIPE_PUBLISHABLE_KEY']}");</script>
+
+= javascript_include_tag 'payment'
```

Comment:
I think you will have to add this `payment.js` file to the asset pipeline. In `config/initializers/assets.rb` there is a comment which you can follow.